system/bin/chroot-distro

```diff
@@ -561,23 +561,27 @@ chroot_distro_download() {
                 1) rootfs="full";rootfs_version="kali-2024.1"; break;;
                 2) rootfs="minimal";rootfs_version="kali-2024.1"; break;;
                 3) rootfs="nano";rootfs_version="kali-2024.1"; break;;
-                4) rootfs="full";rootfs_version="kali-2023.4"; break;;
-                5) rootfs="minimal";rootfs_version="kali-2023.4"; break;;
-                6) rootfs="nano";rootfs_version="kali-2023.4"; break;;
-                7) rootfs="full";rootfs_version="kali-2023.3b"; break;;
-                8) rootfs="minimal";rootfs_version="kali-2023.3b"; break;;
-                9) rootfs="nano";rootfs_version="kali-2023.3b"; break;;
-                10) rootfs="full";rootfs_version="kali-2023.3"; break;;
-                11) rootfs="minimal";rootfs_version="kali-2023.3"; break;;
-                12) rootfs="nano";rootfs_version="kali-2023.3"; break;;
+                4) rootfs="full";rootfs_version="kali-2024.2"; break;;
+                5) rootfs="minimal";rootfs_version="kali-2024.2"; break;;
+                6) rootfs="nano";rootfs_version="kali-2024.2"; break;;
+                7) rootfs="full";rootfs_version="kali-2024.3"; break;;
+                8) rootfs="minimal";rootfs_version="kali-2024.3"; break;;
+                9) rootfs="nano";rootfs_version="kali-2024.3"; break;;
+                10) rootfs="full";rootfs_version="kali-2024.4"; break;;
+                11) rootfs="minimal";rootfs_version="kali-2024.4"; break;;
+                12) rootfs="nano";rootfs_version="kali-2024.4"; break;;
                 13) rootfs="full";rootfs_version="current"; break;;
                 14) rootfs="minimal";rootfs_version="current"; break;;
                 15) rootfs="nano";rootfs_version="current"; break;;
                 *) echo "Unknown option : $number";;
             esac
         done
         if [ ! -f "$chroot_distro_path/.rootfs/$1.tar.xz" ]; then
-            wget -O "$chroot_distro_path/.rootfs/$1.tar.xz" "http://kali.download/nethunter-images/$rootfs_version/rootfs/kalifs-$hardware_machine-$rootfs.tar.xz"
+            if [ "$rootfs_version" = "current" ]; then
+                wget -O "$chroot_distro_path/.rootfs/$1.tar.xz" "http://kali.download/nethunter-images/$rootfs_version/rootfs/kali-nethunter-rootfs-$rootfs-$hardware_machine.tar.xz"
+                else
+                    wget -O "$chroot_distro_path/.rootfs/$1.tar.xz" "http://old.kali.org/nethunter-images/$rootfs_version/rootfs/kalifs-$hardware_machine-$rootfs.tar.xz"
```

Comment:
No need to leave the old format, lets support only the new format. Thus there is no need to if-else branching. This also means that 2024.2 is out of picture as it uses different format.

Reply:
Yes, but there has been change since 2024.3 about how the archive has been formatted, and your patch does not take that in to account.